### DISCUSSION TITLE
Default interpreter setting

### DISCUSSION THREAD

**Discussion by mattkerlogue** [posit-dev/positron#5795] [Feedback] [answered]
Apologies if this is a duplicate, I searched both issues and discussions but couldn't find anything similar.

Would it be possible to add a setting within say `User > Position > Interpreters` to set a default/fallback interpreter (e.g. none (the default), Python, R)?

For existing projects I'm aware Positron makes an informed choice about whether to start in R or Python. But I'm thinking about when I open a new window/session that's not linked to a project/folder.

I noted the following on the [Positron](https://positron.posit.co/interpreters.html) website:

> Positron tries to start R for R users and Python for Python users by checking to see whether RStudio is installed and has been recently used. The latter check is done by looking in ~/.local/share/rstudio for files written less than a month ago, so if you don’t want Positron to consider you to be an RStudio user, you can remove this folder or rename it temporarily.

The issue for me however is that I've just had to have our corporate IT team do a rebuild of my laptop due to an unrelated issue and so I'm having to reinstall software etc. After a near 6-months of solely using Positron I'm not intending (at this stage) to install/open RStudio on the machine and so don't currently have a `~/.local/share/rstudio` folder.

As primarily a R-based user it would be nice to have a setting within Positron itself to set a default/fallback interpreter so that in a new/unattached session it automatically starts up with the interpreter of my choice.

**Comment by mattkerlogue** [comment]
Yes, this is just about me being lazy and not wanting to do two mouse clicks 🤣

**Comment by juliasilge** [comment, marked as answer]
Thank you for this report! 🙌 I think we will solve this category of problem with https://github.com/posit-dev/positron/issues/1181; take a look and see if you have any additional info you'd like to add there. It sounds like this is mostly about _new_ workspaces that don't yet have signals that R should start?

If you are only an R user (never want Python to start) you can [disable the Python extension](https://github.com/posit-dev/positron/issues/3575#issuecomment-2176591849) if you like.

**Comment by mattkerlogue** [comment]
Thanks. Will respond in that issue.

I do occasionally work with Python based projects, so I don't think I'll go as far as disabling the extension.
